Formula/neovim.rb

```diff
@@ -0,0 +1,117 @@
+class Neovim < Formula
+  desc "Ambitious Vim-fork focused on extensibility and agility"
+  homepage "https://neovim.io"
+
+  stable do
+    url "https://github.com/neovim/neovim/archive/v0.2.0.tar.gz"
+    sha256 "72e263f9d23fe60403d53a52d4c95026b0be428c1b9c02b80ab55166ea3f62b5"
+    resource "luarocks" do
+      url "https://github.com/luarocks/luarocks/archive/5d8a16526573b36d5b22aa74866120c998466697.tar.gz"
+      sha256 "cae709111c5701235770047dfd7169f66b82ae1c7b9b79207f9df0afb722bfd9"
+    end
+  end
+
+  head do
+    url "https://github.com/neovim/neovim.git", :shallow => false
+    resource "luarocks" do
+      url "https://github.com/luarocks/luarocks/archive/2.4.2.tar.gz"
+      sha256 "eef88c2429c715a7beb921e4b1ba571dddb7c74a250fbb0d3cc0d4be7a5865d9"
+    end
+  end
+
+  option "with-dev", "Compile a Dev build. Enables debug information, logging and optimizations that don't interfere with debugging."
+
+  depends_on "cmake" => :build
+  depends_on "libtool" => :build
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "pkg-config" => :build
+  depends_on "jemalloc" => :recommended
+  depends_on "libuv"
+  depends_on "msgpack"
+  depends_on "unibilium"
+  depends_on "libtermkey"
+  depends_on "libvterm"
+  depends_on "gettext"
+  depends_on :python => :recommended if MacOS.version <= :snow_leopard
+
+  resource "luv" do
+    url "https://github.com/luvit/luv/archive/1.9.1-0.tar.gz"
+    sha256 "86a199403856018cd8e5529c8527450c83664a3d36f52d5253cbe909ea6c5a06"
+  end
+
+  resource "luajit" do
+    url "https://raw.githubusercontent.com/neovim/deps/master/opt/LuaJIT-2.0.4.tar.gz"
+    sha256 "620fa4eb12375021bef6e4f237cbd2dd5d49e56beb414bee052c746beef1807d"
+  end
+
+  resource "luarocks" do
+    url "https://github.com/luarocks/luarocks/archive/v2.4.2.tar.gz"
+    sha256 "eef88c2429c715a7beb921e4b1ba571dddb7c74a250fbb0d3cc0d4be7a5865d9"
+  end
+
+  def install
+    ENV["HOME"] = buildpath
+
+    resources.each do |r|
+      r.stage(buildpath/"deps-build/build/src/#{r.name}")
+    end
+
+    cd "deps-build" do
+      ohai "Building Neovim third-party dependencies."
+      system "cmake", "../third-party",
+            "-DUSE_BUNDLED_BUSTED=OFF",
```

Comment:
I'm assuming the reason its not done like that now is because of luarocks, luajit and luv since they need to be ON as they are downloaded as separate resources and don't exist as a Formula in homebrew.